### PR TITLE
Map - Improve flashlight support

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- add support for not configured 3rd-party flashlights;
- add support for mounted flashlight;
- add support for weapons with unmountable flashlight (e.g. APEX `arifle_AKM_FL_F` and some CUP rifles);
- fix map flashlight sound issues:
  - no click when flashlight is enabled and map is being closed;
  - click when Chemlight Shield is enabled and map is being opened.

### CODE REVIEW

File: addons/map/functions/fnc_getUnitFlashlights.sqf [hunk 1]
```diff
@@ -20,9 +20,19 @@ params ["_unit"];
 private _flashlights = [];
 
 {
-    if (isText (configFile >> "CfgWeapons" >> _x >> "ItemInfo" >> "FlashLight" >> "ACE_Flashlight_Colour")) then {
-        _flashlights pushBackUnique _x;
-    };
-} forEach (_unit call EFUNC(common,uniqueItems));
+    private _item = _x;
+    private _weaponConfig = configFile >> "CfgWeapons" >> _item;
```
Comment (dedmen) [inline addons/map/functions/fnc_getUnitFlashlights.sqf]: could store `configFile >> "CfgWeapons"` outside of loop for performance

File: addons/map/functions/fnc_getUnitFlashlights.sqf [hunk 2]
```diff
@@ -18,16 +18,17 @@
...
     private _item = _x;
-    private _weaponConfig = configFile >> "CfgWeapons" >> _item;
+    private _weaponConfig = _cfgWeapons >> _item;
     {
         if (
             isText (_x >> "ACE_Flashlight_Colour")
             || {!(getArray (_x >> "ambient") in [[], [0,0,0]])}
-        ) then {
-            _flashlights pushBackUnique _item;
+        ) exitWith {
```
Comment (dedmen) [inline addons/map/functions/fnc_getUnitFlashlights.sqf]: sounds like you want `findIf` here

Comment (Dystopian) [reply addons/map/functions/fnc_getUnitFlashlights.sqf]: yeah but then decided `exitWith` is more logical here.

Comment (dedmen) [reply addons/map/functions/fnc_getUnitFlashlights.sqf]: but less efficient and IMO less clear? Atleast from my perspective

Comment (Dystopian) [reply addons/map/functions/fnc_getUnitFlashlights.sqf]: I only got you now. Can't say it's way more clear but should be faster.

File: addons/map/XEH_postInitClient.sqf [hunk 3]
```diff
@@ -96,15 +96,16 @@ call FUNC(determineZoom);
                 params ["_player", "_mapOn"];
                 private _unitLight = _player getVariable [QGVAR(flashlight), ["", objNull]];
                 _unitLight params ["_flashlight", "_glow"];
-                private _flashlightOn = !(_flashlight isEqualTo "");
                 if (_mapOn) then {
+                    private _flashlightOn = !(_flashlight isEqualTo "");
```
[the remark this answers is not in the document]
Comment (Dystopian) [reply addons/map/XEH_postInitClient.sqf]: I think it was before patch to simplify `if` condition. I'll change it.

**Pull request merged**